### DISCUSSION TITLE
Tasmota

### DISCUSSION THREAD

**Discussion by stefan63** [Arksine/moonraker#212] [Feature Requests]
For Tasmota change to device="tasmota_plug" at least it workt for me. /Stefan

REF:
# printer.cfg
[gcode_macro POWER_OFF_PRINTER]
gcode:
  {action_call_remote_method("set_device_power",
                             `device="printer",`
                              device="tasmota_plug",
                             state="off")}

**Comment by nickeeey** [comment]
thanks. Do you use the printer.cfg for Tasmota? 
i got this in Moonraker:
[power tasmota_plug]
type: tasmota
address: 192.168.0.112
#   The type of device.  Can be either gpio, rf, tplink_smartplug, tasmota
#   shelly, homeseer, homeassistant, loxonev1, or mqtt.
#   This parameter must be provided.
off_when_shutdown: True
....
Controling it with the buttons at the power menu works fine but it didn't stop after printing

**Reply by stefan63** [reply]
Yes, I have the same setup in moonraker as you show.
And the added the macros in Fluidd printer conf.
